### DISCUSSION TITLE
Feature Request: make vcpkg embeddable in docker image.

### DISCUSSION THREAD

**Discussion by vlj** [microsoft/vcpkg#20698] [Q&A]
Currently vcpkg is not automatable : it prompts user before downloading various package like cmake or git and I didn't find a cli argument to always assume "yes". On the other hand docker image building process doesn't allow for user input which makes vcpkg installation impossible in a docker image as part of a "docker build" command.

I'm interested in having a docker image (based on windowsservercore) filled with various build tools to test build instruction from an almost vanilla dev installation.

**Comment by ras0219-msft** [comment]
We have not documented this sufficiently, but you can automatically accept all prompts by creating a file inside `downloads\` called `AlwaysAllowEverything`. It looks like someone else was successful in using us in docker here [1], which might help.

Let us know if you have any other issues; being usable in a docker container is definitely a goal!

[1] https://github.com/dicta/docker-configs/commit/384b5d7133413a5875111aafbec8fda0c1476250

**Comment by vlj** [comment]
there is 2 issues : one is that msbuild complain about missing sdk 10.0.240.0 (although it's installed as part of vs build tool 2015) when executing bootstrap.ps1 . It looks like it's a recent regression, a9f7fc6e90feaad50c1221ef9bd56e2620302215 worked.

Second one is that the Expand-ZIPFile  function doesn't work when installing port, Something related to a nullptr. I will provide more information later when I'm getting acccess to my win 10 pro machine back

**Comment by alexkaratarakis** [comment]
@vlj 
I just pushed cfd5adaf135ccfe2ade3bce43a00d27abd007b0a which improves the Win10 SDK detection.
Does it resolve the first issue?

**Comment by vlj** [comment]
Yes, I can confirm that with master the sdk is properly detected and vcpkg properly build.

**Comment by vlj** [comment]
With the following dockerfile (which embeds both msbuild and vcpkg docker img from dicta config files) :
https://gist.github.com/vlj/95e53e07c9f1fbd7bd4ff5120c7c6987

I have the following error :
https://gist.github.com/vlj/d073cc358eabf95338eee30874f34cec

**Comment by alexkaratarakis** [comment]
@vlj 
03b65486536fc36889a9225fb8dc9de10f823ac6 may fix this (not the MinGit part of the commit, but the change in the Expand-ZIPFile function).

**Comment by vlj** [comment]
It still doesn't work :
https://gist.github.com/vlj/a839f593af2faca17a731263f9f39fd6

**Comment by curtnichols** [comment]
FWIW, I ran into `Launching powershell failed or was denied` running under a build server on Windows (and ended up on this issue via one of @vlj's gist mentioned above; added the error message here for searchability).

This suggestion above worked for me:

> ...you can automatically accept all prompts by creating a file inside `downloads\` called `AlwaysAllowEverything`.

Thanks!

**Comment by KaungZawHtet** [comment]
+1 to this!
vcpkg needs to be extremely small in docker container

**Comment by arthur-tacca** [comment]
I know this is not really the right place for this, but: Has anyone successfully bootstrapped vcpkg and built some ports in a Windows Docker container? If so, would they be kind enough to share their Dockerfile? The original issue that this issue is actually about has been resolved (there's a switch to turn off the prompt) - is there some other problem preventing vcpkg running in Docker?

**Comment by nfrechette** [comment]
I tried and failed. I run into this issue with any package I try to install inside a docker windows image: https://github.com/moby/moby/issues/40266
I can run the image and install the packages fine, but they can't be built into the image which defeats the purpose :(

**Comment by ericoporto** [comment]
@arthur-tacca did you managed to do it? I am trying to figure out a minimal here #14883 but I have a bigger non-minimal there.

**Comment by BillyONeal** [comment]
Now that https://github.com/microsoft/vcpkg/issues/14883 is fixed is this still an issue?

**Comment by badpaybad** [comment]
I can run in docker linux with this dockerfile: https://github.com/badpaybad/cmakes-helloworld/blob/main/Dockerfile 
But image too big, It over 5GB
How to make it smaller?
Can we delete folder vcpkg after we: RUN cmake --build . ???

**Comment by BillyONeal** [comment]
>Can we delete folder vcpkg after we: RUN cmake --build . ???

You can delete `buildtrees` and `packages` but you need `installed`.

**Comment by PhoebeHui** [comment]
Since we don't have docs for this, convert it to discussion.
